apps/dotcom/sync-worker/src/TLUserDurableObject.ts

```diff
@@ -62,17 +61,25 @@ export class TLUserDurableObject extends DurableObject<Environment> {
 			if (rateLimited) {
 				throw new Error('Rate limited')
 			}
-			if (this.lastReplicatorBootId === null) {
-				await this.ctx.blockConcurrencyWhile(async () => {
-					const { data, bootId } = await this.replicator.fetchDataForUser(this.userId!)
-					this.lastReplicatorBootId = bootId
-					this.store.initialize(data)
-					this.replicator.registerUser(this.userId!)
-				})
+			if (this.cache === null) {
+				await this.init()
+			} else {
+				await this.cache.waitUntilConnected()
 			}
 		})
 		.get(`/app/:userId/connect`, (req) => this.onRequest(req))
 
+	private async init() {
+		assert(this.userId, 'User ID not set')
+		this.debug('init')
+		this.cache = new UserDataSyncer(this.ctx, this.env, this.userId, (message) =>
+			this.broadcast(message)
+		)
+		this.debug('cache', !!this.cache)
+		await this.cache.waitUntilConnected()
+		this.replicator.registerUser(this.userId!)
```

Comment:
Looks like both the `TLUserDurableObject` and the `UserDataSyncer` register the user. Not sure if that's intended?